### DISCUSSION TITLE
Packaging data in a single-module distribution

### DISCUSSION THREAD

**Discussion by ericzolf** [pypa/setuptools#4139] [Q&A] [answered]
Hi,

I'm trying to mimic https://github.com/dougn/coverage_pth (abandonware) without the complexity of its setup.py, but I'm failing, the file `coverage.pth` is included in the sdist but never in the wheel.

I've tried multiple parameters in `pyproject.toml` and structures, with and without `src`, with and without `MANIFEST.in`, etc but no avail.

My latest try is available at https://github.com/rdiff-backup/rdiff-backup/pull/952/files

I would interpret `coverage.pth` as a data file in the "root" package, but setuptools seems to have a different interpretation :shrug:

**Comment by abravalheri** [comment, marked as answer]
Files that are not contained by a package folder are not considered package data (they are outside of any package after all).

To add a `.pth` file to a wheel it is necessary to customise the build steps (probably `bdist_wheel` and/or `build_py`, maybe others...).
That is done via `setup.py`. Reminder that `setup.py` is a perfect valid, ***non***-deprecated[^1], configuration/customisation file that can be used alongside/complementary to `pyproject.toml`.

[^1]: The only thing that is deprecated is running this file as a CLI tool.

**Reply by ericzolf** [reply]
I had hoped for a declarative approach to the issue, but fair enough, I'll find an alternative. Actually, I think I'll go back to my scripted approach, finding the first writable directory in `sys.path` to drop the pth file (resp. use the `site` module to do something similar).
